### PR TITLE
Test CWL secondary file from s3.

### LINKED ISSUE #3109: Add S3 support for CWL conformance tests
We currently run CWL conformance tests using different batch systems, but not with different job stores. This can result in bugs like https://github.com/DataBiosphere/toil/issues/3107

We should at least add S3 jobstore support to the CWL conformance tests.

┆Issue is synchronized with this [Jira Task](https://ucsc-cgl.atlassian.net/browse/TOIL-583)
┆Issue Number: TOIL-583

### PR DESCRIPTION
**Pull request by DailyDreaming**
Resolves #3109 .

### CODE REVIEW

**Comment by DailyDreaming** [comment]
@arostamianfar Is this what you had in mind for #3109 / #3107?

**Comment by mr-c** [comment]
@DailyDreaming Is there a test coverage report?

**Comment by DailyDreaming** [comment]
@mr-c Hmmm... reworking the Makefile and the gitlab config to include those again.

**Comment by DailyDreaming** [comment]
@mr-c For `cwltoil.py`, it looks like we have:

```
src/toil/cwl/cwltoil.py                                    873    136    84%
```

When running the CWL 1.0 tests at least.  Coverage is split up though, so each CI job only reports the coverage for the subset of tests that it runs.  We'd have to collect all of the coverage results in one place and somehow sum them up overall, or run the gamut of `make test` on occasion if we want a comprehensive coverage report.

**Comment by arostamianfar** [comment]
Thanks, @DailyDreaming ! this is great! Although, the test I had in mind was to run `cwltoil` with `--jobStore` option that's on S3. I believe the current test ends up downloading the file into a local job store, so it may not necessarily catch the original bug.
Not sure how feasible it is to add such an option?

Regardless, this is a great test to have! Thanks so much!

**Comment by DailyDreaming** [comment]
@arostamianfar Ah, that makes sense.  I just amended it to run with an s3 jobstore.  It passed for me locally so hopefully it runs through gitlab with no problems.

I ended up with a bunch of refactor cruft tacked on after re-enabling coverage.  I think I'm going to break that off into a separate PR and merge it separately.

**Comment by mr-c** [comment]
@DailyDreaming Does the test result in these lines of `cwltool` being executed?

https://github.com/common-workflow-language/cwltool/pull/1319/files#diff-237bb277b731e18286d6ab18769c0991ad2fc6aa0ef4fb64a5f9adeb3034f319R425

**Comment by DailyDreaming** [comment]
@mr-c I just tested it and it does.

File: src/toil/batchSystems/options.py [hunk 1]
```diff
@@ -1,4 +1,3 @@
-from __future__ import absolute_import
 # Copyright (C) 2015-2016 Regents of the University of California
```
Comment (mr-c) [inline src/toil/batchSystems/options.py:1]: no 2017-2020 ?

**Review by mr-c — APPROVED** [review]

File: src/toil/batchSystems/options.py (hunk 1 above)
Comment (DailyDreaming) [reply src/toil/batchSystems/options.py:1]: Ha, will update, thanks!

**Pull request merged**